pyccel/ast/datatypes.py

```diff
@@ -77,23 +78,24 @@
 )
 
 #==============================================================================
-default_precision = {'real': 8, 'int': 8, 'complex': 8, 'bool':4, 'float':8}
-dtype_and_precision_registry = { 'real':('real',8),
-                                 'double':('real',8),
-                                 'float':('real',8),       # sympy.Float
-                                 'pythonfloat':('real',8), # built-in float
+
+default_precision = {'real': 8, 'int': numpy.dtype(int).alignment, 'complex': 8, 'bool':4, 'float':8}
+dtype_and_precision_registry = { 'real':('real',default_precision['float']),
+                                 'double':('real',default_precision['float']),
+                                 'float':('real',default_precision['float']),       # sympy.Float
+                                 'pythonfloat':('real',default_precision['float']), # built-in float
                                  'float32':('real',4),
                                  'float64':('real',8),
-                                 'complex':('complex',8),
+                                 'complex':('complex',default_precision['complex']),
                                  'complex64':('complex',4),
                                  'complex128':('complex',8),
                                  'int8' :('int',1),
                                  'int16':('int',2),
                                  'int32':('int',4),
                                  'int64':('int',8),
-                                 'int'  :('int',8),
+                                 'int'  :('int', default_precision['int']),
                                  'integer':('int',4),
```

Comment:
When is `integer` used?

Reply:
it is also used everywhere we probably need to set its precision to default_precision

Reply:
I can't find it used anywhere

Reply:
when you use the function `str_dtype` it's converted to it

Reply:
yes we can

Reply:
I would also prefer to remove `integer` and use `int` everywhere, unless there is a good reason to keep both.

Reply:
Clean up of the dictionaries in `pyccel.ast.datatypes` should be done when we work on issue #305.